osmomath/binary_search.go

```diff
@@ -141,6 +141,63 @@ func (e ErrTolerance) CompareBigDec(expected BigDec, actual BigDec) int {
 	return 0
 }
 
+// CompareDec validates if actual is within errTolerance of expected.
+// returns 0 if it is
+// returns 1 if not, and expected > actual.
+// returns -1 if not, and expected < actual
```

Comment:
Can we add more test cases for all the logical branch in this method?
e.g) 
- it can either be round up or round down 
- it can either be additive or multiple tolerance.


And so on. I think we can make tdd depending on different combinations of paramters? Wdyt?

Reply:
> I think we can make tdd depending on different combinations of paramters? 

yes, there already existed a [test](https://github.com/osmosis-labs/osmosis/blob/ea8307152112afd5805c455057a50e94ed265b1d/osmomath/binary_search_test.go#L321-L372) for `Int` and `BigDec` comparison prior to this PR, which includes different configurations you mention. I added `Dec` comparison to that test in this commit: https://github.com/osmosis-labs/osmosis/pull/6336/commits/d9f07a45ce956ab92f4816b6ad29e8e478451fde